modules/core/src/v2/coins/algo.ts

```diff
@@ -1,35 +1,115 @@
 import BaseCoin = require('../baseCoin');
+const algosdk = require('algosdk');
+
+interface KeyPair {
+  pub: string;
+  prv: string;
+  secret: string;
+}
 
 class Algo extends BaseCoin {
+  constructor() {
+    super();
+  }
 
-    constructor() {
-      super();
-    }
+  getChain(): string {
+    return 'algo';
+  }
 
-    getChain() {
-        return 'algo';
-    }
+  getFamily(): string {
+    return 'algo';
+  }
 
-    getFamily() {
-        return 'algo';
-    }
+  getFullName(): string {
+    return 'Algorand';
+  }
 
-    getFullName() {
-        return 'Algorand';
-    }
-
-    getBaseFactor() {
-        return 1e6;
-    }
+  getBaseFactor(): number {
+    return 1e6;
+  }
 
-    /**
+  /**
      * Flag for sending value of 0
      * @returns {boolean} True if okay to send 0 value, false otherwise
      */
-    valuelessTransferAllowed(): boolean {
-        // TODO: this sounds like its true with the staking txes - confirm before launch
-        return false;
+  valuelessTransferAllowed(): boolean {
+    // TODO: this sounds like its true with the staking txes - confirm before launch
+    return false;
+  }
+
+  /**
+   * Generate ed25519 key pair
+   *
+   * @param seed
+   * @returns {Object} object with generated pub, prv and secret
+   */
+  generateKeyPair(seed?: Buffer): KeyPair {
+    const pair = seed ? algosdk.generateAccountFromSeed(seed) : algosdk.generateAccount();
+    return {
+      pub: pair.addr, // encoded pub
+      prv: algosdk.Seed.encode(pair.sk), // encoded seed
```

Comment:
Why are we returning the encoded seed as part of the keypair? This breaks the basecoin contract by returning a different object than other coins.

Seems like you can go back and forth from the seed to the pub/prv, so why do  we need to keep the seed after we generate the keypair?

Reply:
Is there any code which needs to use `secret`? The tx signing code recovers it from `prv` instead of getting it from the keypair: https://github.com/BitGo/BitGoJS/pull/368/files#diff-32749934abcd4fda4b31a0f6496db33aR105

Reply:
yes, `signMessage`: https://github.com/BitGo/BitGoJS/pull/367/files#diff-32749934abcd4fda4b31a0f6496db33aR111

Reply:
my bad, we can regenerate the key with the seed using `algosdk.generateAccountFromSeed` and use `key.pk` for signing. Let me fix that.

Reply:
done